### PR TITLE
Modify strictness of color parser (#357)

### LINKED ISSUE #357: Color parser is both strict and loose
The following styles are *generally* valid according to the CSS Level 3, but are rejected in Colors.jl.
```julia
julia> parse(RGBA,"rgba(1,10,100,.5)")
ERROR: Unknown color: rgba(1,10,100,.5)

julia> parse(RGBA,"rgba(1,10,100,\t0.5)")
ERROR: Unknown color: rgba(1,10,100,       0.5)
```

Although we do not have to support parsing the all numerical notations, I think it is worth supporting the former because the former is shorter than something like `+5e-1`.

On the other hand, the following styles are accepted without errors or warnings.
```julia
julia> parse(Colorant,"#FF8000demo")
RGB{N0f8}(1.0,0.502,0.0)

julia> parse(RGB,"p ink")
RGB{N0f8}(1.0,0.753,0.796)
```

The former is potentially dangerous since Colors.jl does not support parsing the 8-digit or 4-digit hex notation yet, as mentioned in #353.
The latter is not so dangerous, but the `replace()` is slightly slow.
https://github.com/JuliaGraphics/Colors.jl/blob/d64fb1a6bd88c4374d72574744c27a7539a4a8ca/src/parse.jl#L50
In the cases where the regex matching is used, there is no need to `replace` the input string. In the case of the named colors, `strip()`, which returns `SubString`, may be helpful.

### PR DESCRIPTION
**Pull request by kimikage**
This fixes #357.
- Modify white space handling
- Add error message about 8-digit or 4-digit hex notation
- Add support for some CSS4 notations
- Add tests
- Speed up parsing averagely

### CODE REVIEW

**Comment by kimikage** [comment]
It is not difficult to implement the 8-digit or 4-digit hex notation, but we should settle the issue #353.

File: test/parse.jl [hunk 1]
```diff
@@ -4,23 +4,46 @@ using FixedPointNumbers
...
+    @test parse(Colorant, "\nSeaGreen ") === RGB{N0f8}(r8(0x2E),r8(0x8B),r8(0x57))
+
+    # hex-color
     @test parse(Colorant, "#D0FF58") === RGB(r8(0xD0),r8(0xFF),r8(0x58))
+    @test parse(Colorant, "0xd0ff58") === RGB(r8(0xD0),r8(0xFF),r8(0x58))
     @test parse(Colorant, "#FB0") === RGB(r8(0xFF),r8(0xBB),r8(0x00))
+    @test_throws ErrorException parse(Colorant, "#FB0A")
+    @test_throws ErrorException parse(Colorant, "#BAD05")
+    @test_throws ErrorException parse(Colorant, "#BAD0007")
+    @test_throws ErrorException parse(Colorant, "#FFBB00AA") # not supported yet
```
Comment (timholy) [inline test/parse.jl]: In such cases is `@test_broken` more appropriate?

Comment (kimikage) [reply test/parse.jl]: That's right, if you have already specified the *correct* behavior. Throwing the "not supported" error is not a *broken* behavior *now*. (Of course, checking the error message might be *better*.)

Comment (timholy) [reply test/parse.jl]: I'm fine with it either way, no reason not to merge.

**Comment by kimikage** [comment]
If this PR causes a problem (e.g. "not supported" error), it is probably a potential bug in the user scripts.
Although this behavior is going to be changed soon, but I will merge this foremost.

**Comment by kimikage** [comment]
Aww, I'm sorry but I mistaken this PR number and the issue number.

**Pull request merged**